src/test/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListenerTest.java

```diff
@@ -165,32 +198,56 @@ public void testOnCompletedSuccessWithStatsD() throws Exception {
         envVars.put("BUILD_URL", "http://build_url.com");
         envVars.put("GIT_BRANCH", "test-branch");
 
-        Run run = mock(Run.class);
-        when(run.getResult()).thenReturn(Result.SUCCESS);
-        when(run.getEnvironment(any(TaskListener.class))).thenReturn(envVars);
-        when(run.getDuration()).thenReturn(1234000L); // pipeline jobs always return 0
-        when(run.getNumber()).thenReturn(2);
-        when(run.getParent()).thenReturn(job);
+        Run runFailures = mock(Run.class);
+        when(runFailures.getResult()).thenReturn(Result.FAILURE);
+        when(runFailures.getEnvironment(any(TaskListener.class))).thenReturn(envVars);
+        when(runFailures.getDuration()).thenReturn(123000L);
+        when(runFailures.getNumber()).thenReturn(2);
+        when(runFailures.getParent()).thenReturn(job);
+        when(datadogBuildListener.getMeanTimeBetweenFailure(runFailures)).thenReturn(123000L);
+
+        datadogBuildListener.onCompleted(runFailures, mock(TaskListener.class));
 
-        datadogBuildListener.onCompleted(run, mock(TaskListener.class));
         String[] expectedTags = new String[4];
         expectedTags[0] = "job:ParentFullName/JobName";
         expectedTags[1] = "node:test-node";
-        expectedTags[2] = "result:SUCCESS";
+        expectedTags[2] = "result:FAILURE";
         expectedTags[3] = "branch:test-branch";
+        client.assertMetric("jenkins.job.duration", 123, "null", expectedTags);
+        client.assertMetric("jenkins.job.feedbacktime", 123, "null", expectedTags);
+        client.assertMetric("jenkins.job.mtbf", 123, "null", expectedTags);
+        client.assertServiceCheck("jenkins.job.status", 2, "null", expectedTags);
+    }
 
-        statsd.assertMetric("completed", 1, expectedTags);
-        statsd.assertMetric("leadtime", 1234, expectedTags);
-        statsd.assertedAllMetricsAndCounters();
+    @Test
+    public void testJenkinsRuns() throws Exception {
```

Comment:
I am not sure to understand what this test does.

Reply:
As per your comment above: 

> Also we should go deeper testing mttr, cycleTime, leadTime metrics. Each deserve their own tests.

This is to test that the values we are getting for the methods that generate these variables are not null.